src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/FileSystemApiTest.cs

```diff
@@ -0,0 +1,1013 @@
+            }
+        }
+
+        [Fact]
+        public void AddFile_Large()
+        {
+            AddFile(); // warm up
+
+            var path = "star_trails.mp4";
+            var stopWatch = new Stopwatch();
+            stopWatch.Start();
+            var node = ipfs.UnixFsApi.AddFileAsync(path).Result;
+            stopWatch.Stop();
+            
+            // _testOutputHelper.WriteLine("Add file took {0} seconds.", stopWatch.Elapsed.TotalSeconds);
+
+            Assert.Equal("QmeZkAUfUFPq5YWGBan2ZYNd9k59DD1xW62pGJrU3C6JRo", node.Id);
+
+            var k = 8 * 1024;
+            var buffer1 = new byte[k];
+            var buffer2 = new byte[k];
+            stopWatch.Restart();
+            using (var localStream = new FileStream(path, FileMode.Open, FileAccess.Read))
+            {
+                using var ipfsStream = ipfs.UnixFsApi.ReadFileAsync(node.Id).Result;
+                while (true)
+                {
+                    var n1 = localStream.Read(buffer1, 0, k);
+                    var n2 = ipfsStream.Read(buffer2, 0, k);
+                    Assert.Equal(n1, n2);
+                    if (n1 == 0)
+                    {
+                        break;
+                    }
+                    
+                    for (var i = 0; i < n1; ++i)
+                    {
+                        if (buffer1[i] != buffer2[i])
+                        {
+                            throw new XunitException("data not the same");
+                        }
+                    }
+                }
+            }
+
+            stopWatch.Stop();
+            
+            // _testOutputHelper.WriteLine("Readfile file took {0} seconds.", stopWatch.Elapsed.TotalSeconds);
+        }
+
+        /// <seealso href="https://github.com/richardschneider/net-ipfs-engine/issues/125"/>
+        [Fact]
+        public void AddFile_Larger()
+        {
+            AddFile(); // warm up
+
+            var path = "starx2.mp4";
+            var stopWatch = new Stopwatch();
+            stopWatch.Start();
+            var node = ipfs.UnixFsApi.AddFileAsync(path).Result;
+            stopWatch.Stop();
+            _testOutputHelper.WriteLine("Add file took {0} seconds.", stopWatch.Elapsed.TotalSeconds);
+
+            Assert.Equal("QmeFhfB4g2GFbxYb7usApWzq8uC1vmuxJajFpiJiT5zLoy", node.Id);
+
+            const int k = 8 * 1024;
+            var buffer1 = new byte[k];
+            var buffer2 = new byte[k];
+            stopWatch.Restart();
+            using (var localStream = new FileStream(path, FileMode.Open, FileAccess.Read))
+            {
+                using var ipfsStream = ipfs.UnixFsApi.ReadFileAsync(node.Id).Result;
+                while (true)
+                {
+                    var n1 = localStream.Read(buffer1, 0, k);
+                    var n2 = ipfsStream.Read(buffer2, 0, k);
+                    Assert.Equal(n1, n2);
+                    if (n1 == 0)
+                    {
+                        break;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor the code in order to remove this break statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)